### DISCUSSION TITLE
Inpaint - Change character, but keep pose and background

### DISCUSSION THREAD

**Discussion by MarkGree** [Mikubill/sd-webui-controlnet#748] [Q&A]
I want to replace a person in image using inpaint+controlnet openpose

**My workflow:**
1. Set inpaint image, draw mask over character to replace
Masked content: Original
Inpainting area: Only Masked
2. Enable controlnet, set preprocessor & adapter: openpose
3. Generate

- What I get: completely changed image, but with controlnet generated pose.
- What I need: new character replaced previous character **in masked area**, with previous character's pose
So, openpose didn't work with inpaint mask for me

**Comment by marcsyp** [comment]
Make sure you don't have "Skip img2img processing when using img2img initial image" checked in the ControlNet settings -- this prevents inpainting from working properly.

**Comment by ThereforeGames** [comment]
This is precisely what Bodysnatcher is for: https://www.reddit.com/r/StableDiffusion/comments/12o6qit/bodysnatcher_new_fullbody_swapper_just_released/

**Comment by maelisjewelry-commits** [comment]
צור לי תמונה של אדם יושב על ספה זאת שיראה אמיתי 
![b805e685-dc1e-4091-8908-89de6839b442](https://github.com/user-attachments/assets/4210ea22-b554-4022-b066-4ee9a90920f1)
